packages/teleport/src/Apps/AddApp/Automatically/Automatically.tsx

```diff
@@ -0,0 +1,187 @@
+              />
+              <FieldInput
+                rule={requiredAppUri}
+                label="APP URL"
+                width="100%"
+                value={uri}
+                placeholder="https://localhost:4000"
+                onKeyPress={e => handleEnterPress(e, validator)}
+                onChange={e => setUri(e.target.value)}
+              />
+            </Flex>
+            {!cmd && (
+              <Text mb="3">
+                Provide a name and URI of your application to generate an
+                automatic script which you can run on the server that has access
+                to your application
+              </Text>
+            )}
+            {attempt.status === 'failed' && (
+              <Alert kind="danger" children={attempt.statusText} />
+            )}
+            {cmd && (
+              <>
+                <Text mb="3">
+                  Use below script to add an application to your cluster. <br />
+                  The script will be valid for
+                  <Text bold as="span">
+                    {` ${expires}`}.
+                  </Text>
+                </Text>
+                <TextSelectCopy text={cmd} mb={2} />
+              </>
+            )}
+          </DialogContent>
+          <DialogFooter>
+            {!cmd && (
+              <ButtonPrimary
+                mr="3"
+                disabled={attempt.status === 'processing'}
+                onClick={() => handleCreate(validator)}
+              >
+                Generate Script
+              </ButtonPrimary>
+            )}
+            {cmd && (
+              <ButtonPrimary
+                mr="3"
+                disabled={attempt.status === 'processing'}
+                onClick={() => handleCreate(validator)}
+              >
+                Re-generate
+              </ButtonPrimary>
+            )}
+            <ButtonSecondary
+              disabled={attempt.status === 'processing'}
+              onClick={onClose}
+            >
+              Close
+            </ButtonSecondary>
+          </DialogFooter>
+        </>
+      )}
+    </Validation>
+  );
+}
+
+const requiredAppUri = value => () => {
+  if (!value || value.length === 0) {
+    return {
+      valid: false,
+      message: 'Required',
+    };
+  }
+
+  try {
+    new URL(value);
+  } catch (err) {
+    return {
+      valid: false,
+      message: 'URL is invalid',
+    };
+  }
+
+  return {
+    valid: true,
+  };
+};
+
+const requiredAppName = value => () => {
+  if (!value || value.length === 0) {
+    return {
+      valid: false,
+      message: 'Required',
+    };
+  }
+
+  try {
+    const tmp = new URL(`https://${value}`);
+    if (tmp.hostname !== value) {
+      throw new Error();
+    }
+
+    // cannot be a sub-domain
+    if (tmp.hostname.split('.').length > 1) {
+      throw new Error();
+    }
+  } catch (err) {
+    return {
+      valid: false,
+      message: 'Invalid',
+    };
+  }
```

Comment:
For this test, are we trying to catch for like spaces, dots, and invalid characters by taking advantage of new URL() validation?

Reply:
Correct

Reply:
This breaks on `http://username:password@subdomain.example.com` which is a legitimate URL - the script outputs:

`sudo bash -c "$(curl -fsSL 'https://teleport.vagrant:3080/scripts/417512bcdab893fd596e73f819387d91/install-app.sh?name=gus-test-app&uri=http://subdomain.example.com')"` where the `username:password@` is missing.

Reply:
@webvictim you found one :) I will address it by encoding entire URL as promised before merging to master (still stratus).

Reply:
Thanks @alex-kovoy!